iroh-net/src/magicsock.rs

```diff
@@ -345,9 +348,18 @@ impl Inner {
             }
             n += 1;
         }
-        // TODO: This is the remaining alloc on the hot path for send.
-        // Unfortunately I don't see a way around this because we have do modify the transmits.
-        let mut transmits = transmits[..n].to_vec();
+
+        // Copy the transmits into an owned buffer, because we will have to modify the send
+        // addresses to translate from the quic mapped address to the actual UDP address.
+        // To avoid allocating on each call to `poll_send`, we use a fixed buffer.
+        let mut transmits = {
+            let mut buf = self.send_buffer.lock();
+            buf.clear();
```

Comment:
no need to clear, you can just overwrite

Reply:
how do I overwrite when the new length is greater than the old length?